app/models/reservation.rb

```diff
@@ -60,7 +71,7 @@ def self.joins_order
       .joins("LEFT JOIN orders ON orders.id = order_details.order_id")
   end
 
-  scope :not_canceled, -> { where(canceled_at: nil) }
+  scope :not_canceled, -> { where(order_details: { canceled_at: nil }) } # dubious
```

Comment:
Yeah, the way this ends up being used are generally in some pretty complicated queries that involve a lot of joins. Might be good to pair on finding a better solution.